src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -100,6 +101,35 @@ export interface ITerminalInstanceService {
 	didRegisterBackend(backend: ITerminalBackend): void;
 }
 
+/**
+ * Service for bridging terminal chat related tool invocations with terminal instances.
+ * Maintains an in-memory association between a tool session id (generated when a chat tool
+ * prepares/invokes a terminal run) and the actual {@link ITerminalInstance}. Consumers (eg. chat
+ * UI parts) can look up the instance asynchronously or listen for registrations should the
+ * terminal not yet exist.
+ */
```

Comment:
You could simplify this by saying it's mainly a communication mechanism to enable communication via the chat tool implementation in terminalContrib and workbench contribs.

> Maintains an in-memory association between a tool session id

This will probably quickly get out of date as we add functionality.